lib/zooniverse_user.rb

```diff
@@ -82,4 +69,28 @@ def simple_token
   def hex_token
     SecureRandom.hex 64
   end
+
+  def setup_panoptes_user_account(u)
+    panoptes_account_exists = panoptes_user_account_exists?(u)
+    new_account = !u.persisted?
+    u.display_name = login
+    u.email = email
+    u.encrypted_password = crypted_password
+    u.password_salt = password_salt
+    if new_account || panoptes_account_exists
+      u.created_at = created_at
+      u.updated_at = updated_at
+      u.hash_func = 'sha1'
+      u.migrated = true
+      u.zooniverse_id = id.to_s
+      u.build_identity_group if new_account
+    end
+  end
+
+  def panoptes_user_account_exists?(user)
+    user.display_name = login &&
+    user.zooniverse_id.nil? &&
+    user.hash_func == 'bcrypt' &&
```

Comment:
If you're setting the encrypted password to the crypted password from Zoo Home this needs to be 'sha1'. It'll be switched back to bcrypt whenever we enable login through Panoptes.